iib/workers/tasks/utils.py

```diff
@@ -19,11 +19,153 @@
 
 from iib.exceptions import IIBError
 from iib.workers.config import get_worker_config
+from iib.workers.api_utils import set_request_state
 
 log = logging.getLogger(__name__)
 dogpile_cache_region = create_dogpile_region()
 
 
+def get_binary_image_from_config(ocp_version, distribution_scope, binary_image_config={}):
+    """
+    Determine the binary image to be used to build the index image.
+
+    :param str ocp_version: the ocp_version label value of the index image.
+    :param str distribution_scope: the distribution_scope label value of the index image.
+    :param dict binary_image_config: the dict of config required to identify the appropriate
+        ``binary_image`` to use.
+    :return: pull specification of the binary_image to be used for this build.
+    :rtype: str
+    :raises IIBError: when the config value for the ocp_version and distribution_scope is missing.
+    """
+    binary_image = binary_image_config.get(distribution_scope, {}).get(ocp_version, None)
+    if not binary_image:
+        raise IIBError(
+            'IIB does not have a configured binary_image for'
+            f' distribution_scope : {distribution_scope} and ocp_version: {ocp_version}.'
+            ' Please specify a binary_image value in the request.'
+        )
+
+    return binary_image
+
+
+class RequestConfig:
+    """Request config abstract class.
+
+    :param binary_image:  the pull specification of the container image
+                          where the opm binary gets copied from.
+    :type binary_image: str
+    :param overwrite_from_index_token: the token used for overwriting the input
+        ``from_index`` image. This is required for
+        non-privileged users to use ``overwrite_from_index``.
+        The format of the token must be
+        in the format "user:password".
+    :type overwrite_from_index_token: str
+    :param distribution_scope: the scope for distribution
+        of the index image, defaults to ``None``.
+    :type distribution_scope: str
+    :param source_from_index: the pull specification of the container image
+        containing the index that will be used
+        as a base of the merged index image.
+    :type source_from_index: str
+    :param target_index: the pull specification of the container image
+        containing the index whose new data will be added
+        to the merged index image.
+    :type target_index: str
+    :param binary_image_config: the dict of config required to
+        identify the appropriate ``binary_image`` to use.
+    :type binary_image_config: dict
+    """
+
+    _attrs = [
+        "_binary_image",
+        "add_arches",
+        "distribution_scope",
+        "source_from_index",
+        "target_index",
+        "binary_image_config",
+    ]
+    __slots__ = _attrs
+
+    def __init__(self, **kwargs):
+        """Request config __init__.
```

Comment:
could you please start the comment on a new line after `"""`

Reply:
https://www.python.org/dev/peps/pep-0008/#documentation-strings

Reply:
https://github.com/release-engineering/iib#coding-standards

Reply:
@midnightercz

Reply:
Could you please fix the docstring as per https://github.com/release-engineering/iib#coding-standards